geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/SpatialUnits.groovy

```diff
@@ -351,53 +351,85 @@ IProcess createBlocks() {
+ * @param idColumnTab1 The column name where is stored the ID of table 1
+ * @param idColumnTab2 The column name where is stored the ID of table 2
  * @param prefixName A prefix used to name the output table
- * @param nbRelations Number of relations that the lower scale object can have with the upper scale object (i.e. if
- * nbRelations = 1 for buildings and RSU, the buildings can have only one RSU as relation)
+ * @param pointOnSurface Whether or not the spatial join may be performed on pointOnSurfaceTypes of spatial join that may be used (default false):
+ *          --> True: polygons from the first table are converted to points before to be spatially join with polygons from the second table
+ *          --> False: polygons from the first table are directly spatially joined with polygons from the second table
+ * @param nbRelations If 'pointOnSurface' is False, number of relations that one polygon in Table 2 may have with
+ * polygons from Table 1 (e.g. if nbRelations = 1 for buildings and RSU, the buildings can have only one RSU as relation).
+ * The selection of which polygon(s) need to be conserved is based on shared polygons area. By default, this parameter
+ * has no value and thus all relations are conserved
  *
- * @return A database table name and the name of its ID field
+ * @return outputTableName A table name containing ID from table 1, ID from table 2 and AREA shared by the two objects (if pointOnSurface = false)
+ * @return idColumnTab2 The ID name of the table 2
  */
-IProcess createScalesRelations() {
+IProcess spatialJoin() {
     return create {
-        title "Creating the Tables of relations between two scales"
-        id "createScalesRelations"
-        inputs inputLowerScaleTableName: String, inputUpperScaleTableName: String, idColumnUp: String,
-                prefixName: String, nbRelations: 1, datasource: JdbcDataSource
-        outputs outputTableName: String, outputIdColumnUp: String
-        run { inputLowerScaleTableName, inputUpperScaleTableName, idColumnUp, prefixName, nbRelations, datasource ->
+        title "Creating the spatial join between two tables of polygons"
+        id "spatialJoin"
+        inputs inputTable1: String, inputTable2: String, idColumnTab1: String,
+                idColumnTab2: String, prefixName: String, pointOnSurface: false, nbRelations: int,
+                datasource: JdbcDataSource
+        outputs outputTableName: String, idColumnTab2: String
+        run { inputTable1, inputTable2, idColumnTab1, idColumnTab2, prefixName, pointOnSurface,
+              nbRelations, datasource ->
 
             def GEOMETRIC_COLUMN_LOW = "the_geom"
             def GEOMETRIC_COLUMN_UP = "the_geom"
 
-            info "Creating the Tables of relations between the twos scales :  $inputLowerScaleTableName and $inputUpperScaleTableName"
+            info "Creating a spatial join between objects from two tables :  $inputTable1 and $inputTable2"
```

Comment:
Creating a spatial join between geometries from two tables :  $inputTable1 and $inputTable2

Reply:
ok